python/flatbuffers/builder.py

```diff
@@ -514,22 +514,25 @@ def __Finish(self, rootTable, sizePrefix, file_identifier=None):
         """Finish finalizes a buffer, pointing to the given `rootTable`."""
         N.enforce_number(rootTable, N.UOffsetTFlags)
 
+        prepSize = N.UOffsetTFlags.bytewidth
+        if file_identifier is not None:
+            prepSize += encode.FILE_IDENTIFIER_LENGTH
+        if sizePrefix:
+            prepSize += N.Int32Flags.bytewidth
+        self.Prep(self.minalign, prepSize)
+
         if file_identifier is not None:
-            self.Prep(N.UOffsetTFlags.bytewidth, encode.FILE_IDENTIFIER_LENGTH)
-            
             # Convert bytes object file_identifier to an array of 4 8-bit integers,
             # and use big-endian to enforce size compliance.
-            # https://docs.python.org/2/library/struct.html#format-characters
+            # https://docs.python.org/library/struct.html#format-characters
             file_identifier = N.struct.unpack(">BBBB", file_identifier)
-            for i in range(encode.FILE_IDENTIFIER_LENGTH-1, -1, -1):
-                # Place the bytes of the file_identifer in reverse order:
-                self.Place(file_identifier[i], N.Uint8Flags)
+            for char in reversed(file_identifier):
```

Comment:
This removes an implicit assertion that the file_identifier length is correct. Could you add an assert to that effect back in to the code?

Reply:
The assertion isn't necessary. `struct.unpack` errors if the provided value doesn't match the decoding format, which is a 4 byte array.

Reply:
@pyottamus That wasn't obvious to me, and it looks like you're right. Please add a comment documenting that fact.